fastly/pool.go

```diff
@@ -0,0 +1,293 @@
+package fastly
+
+import (
+	"fmt"
+	"net/url"
+	"sort"
+)
+
+const (
+	// PoolTypeRandom is a pool that does random direction.
+	PoolTypeRandom PoolType = "random"
+
+	// PoolTypeHash is a pool that does hash direction.
+	PoolTypeHash PoolType = "hash"
+
+	// PoolTypeClient ins a pool that does client direction.
+	PoolTypeClient PoolType = "client"
+)
+
+// PoolType is a type of pool.
+type PoolType string
+
+// Pool represents a pool response from the Fastly API.
+type Pool struct {
+	ServiceID string `mapstructure:"service_id"`
+	Version   int    `mapstructure:"version"`
+
+	ID               string   `mapstructure:"id"`
+	Name             string   `mapstructure:"name"`
+	Comment          string   `mapstructure:"comment"`
+	Shield           string   `mapstructure:"shield"`
+	RequestCondition string   `mapstructure:"request_condition"`
+	MaxConnDefault   uint     `mapstructure:"max_conn_default"`
+	ConnectTimeout   uint     `mapstructure:"connect_timeout"`
+	FirstByteTimeout uint     `mapstructure:"first_byte_timeout"`
+	Quorum           uint     `mapstructure:"quorum"`
+	UseTLS           bool     `mapstructure:"use_tls"`
+	TLSCACert        string   `mapstructure:"tls_ca_cert"`
+	TLSCiphers       string   `mapstructure:"tls_ciphers"`
+	TLSClientKey     string   `mapstructure:"tls_client_key"`
+	TLSClientCert    string   `mapstructure:"tls_client_cert"`
+	TLSSNIHostname   string   `mapstructure:"tls_sni_hostname"`
+	TLSCheckCert     bool     `mapstructure:"tls_check_cert"`
+	TLSCertHostname  string   `mapstructure:"tls_cert_hostname"`
+	MinTLSVersion    string   `mapstructure:"min_tls_version"`
+	MaxTLSVersion    string   `mapstructure:"max_tls_version"`
+	Healthcheck      string   `mapstructure:"healthcheck"`
+	Type             PoolType `mapstructure:"type"`
+	OverrideHost     string   `mapstructure:"override_host"`
```

Comment:
Please add `created_at`, `updated_at`, and `deleted_at` fields to the response struct. These fields are not listed in the docs but belong to the response body.

Reply:
Fixed in 25093b6